producer/types.go

```diff
@@ -96,9 +107,8 @@ type Buffer interface {
 	// Init initializes the buffer.
 	Init()
 
-	// Close stops the buffer from accepting new requests immediately.
-	// It will also block until all the data buffered has been consumed.
-	Close()
+	// Close closes the buffer.
```

Comment:
Not sure if you feel it is redundant but it might be nice to have the comment here that you have for the Producer close function on line 81.